swiftpkg/internal/swift_registry_package.bzl

```diff
@@ -0,0 +1,74 @@
+"""Implementation for `swift_registry_package`."""
+
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "update_attrs")
+load(":pkg_ctxs.bzl", "pkg_ctxs")
+load(":repo_rules.bzl", "repo_rules")
+load(":repository_files.bzl", "repository_files")
+
+def _remove_bazel_files(repository_ctx, directory):
+    files = ["BUILD.bazel", "BUILD", "WORKSPACE", "WORKSPACE.bazel"]
+    for file in files:
+        repository_files.find_and_delete_files(repository_ctx, directory, file)
+
+def _swift_registry_package_impl(repository_ctx):
+    directory = str(repository_ctx.path("."))
+    env = repo_rules.get_exec_env(repository_ctx)
+    repo_rules.check_spm_version(repository_ctx, env = env)
+
+    url = repository_ctx.attr.url
+
+    # Works for my case, but the prefix can actually be anything, so this is not sufficient for general use.
+    # See `archive-source` command docs:
+    # https://github.com/apple/swift-evolution/blob/main/proposals/0292-package-registry-service.md#archive-source-subcommand
+    components = url.split("/")
+    scope = components[-3]
+    name = components[-2]
+    prefix = "{}.{}".format(scope, name)
+
+    headers = { 
+        "Accept": "application/vnd.swift.registry.v1+zip",
+    }
+
+    repository_ctx.download_and_extract(
+        url = url,
+        # requires bazel 7.1
+        headers = headers,
```

Comment:
This is probably fine for a new feature, we should call this out in any documentation though. I wonder if folks on Bazel < 6 but > 5 could use the credentials helper to work around it?

Reply:
> This is probably fine for a new feature, we should call this out in any documentation though. I wonder if folks on Bazel < 6 but > 5 could use the credentials helper to work around it?

I'm not familiar with the credentials helper; can it override the `Accept` header?

Reply:
I don't like requiring Bazel 7.1. I'm also not familiar with the credentials helper. @luispadron do you know an answer to their question?